src/main/ruby/core/truffle/cext.rb

```diff
@@ -78,11 +78,17 @@ def self.library_in_search_path_or_default(lib, search_paths)
     end
 
     def self.standardize_lib_name(lib)
-      if lib.start_with?('lib') or lib.start_with?('/')
-        lib
-      else
-        "lib#{lib}.#{RbConfig::CONFIG['NATIVE_DLEXT']}"
-      end
+      require 'ffi'
+      extend FFI::Library
+
+      mapped_lib = FFI.map_library_name(lib)
+
+      # Mapping a library name will only handle hard-coded cases accurately. Otherwise, it takes a best effort approach
+      # towards standardizing a library name. On Linux, that name may correspond to an `ld` script, rather than the
+      # actual library we want to load. So, we try loading the library here, which will properly resolve the full
+      # library path if necessary.
+      dynamic_lib = ffi_lib(mapped_lib).first
+      dynamic_lib.name
     end
   end
```

Comment:
Mmh, we should probably move the launcher to its own file, there is no need to have it in core.
(That might be true for this whole file as well, we could require it only once we load `ruby.su`).

Reply:
I filed https://github.com/graalvm/truffleruby/issues/799